main.go

```diff
@@ -370,7 +373,37 @@ func main() {
 		}
 	}
 	// handle profiler sink close on exit
-	defer prfl.Stop()
+	defer func() {
+		logger.Infow("closing profiler sink client before exiting", "sink", prfl.GetSinkName())
+		prfl.Stop()
+	}()
+
+	// tracer sink
+	ts, err := tracer.GetSink(logger, tracertypes.SinkDriver(cfg.Controller.TracerSink))
```

Comment:
Prefer better naming other bad naming + comment
```suggestion
	tracerSink err := tracer.GetSink(logger, tracertypes.SinkDriver(cfg.Controller.TracerSink))
```